File: 251002/Troitsky/Lab1-5/Lab1.Application/Services/UserService.cs
using Lab1.Core.Abstractions;
using Lab1.Core.Models;
using Lab1.Core.Contracts;
using System.Numerics;
using FluentValidation;
using Lab1.Application.Exceptions;
namespace Lab1.Application.Services
{
    public class UserService : IUserService
    {
        private IUserRepository _repository;
        private IValidator<User> _validator;
        public UserService(IUserRepository repository, IValidator<User> validator) => (_repository, _validator) = (repository, validator);
        public UserResponseTo CreateUser(User user)
        {
            var res = _validator.Validate(user);
            if (!res.IsValid)
                throw new IncorrectDataException("invalid input");
            try
            {
                return _repository.Create(user);
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
        public bool DeleteUser(ulong id)
        {
            try
            {
                return _repository.Delete(id);
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
        public UserResponseTo? GetUser(ulong id)
        {
            try
            {
                return _repository.Get(id);
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
        public List<UserResponseTo> GetAllUsers()
        {
            try
            {
                return _repository.GetAll();
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
        public UserResponseTo? UpdateUser(User user, ulong id)
        {
            var res = _validator.Validate(user);
            if (!res.IsValid)
                throw new IncorrectDataException("invalid input");
            try
            {
                return _repository.Update(user, id);
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
    }
}

File: 251001/Karpekov/lab1/TaskRest/src/main/java/lab1/services/UserService.java
package lab1.services;

import lab1.datalayer.DataLayer;
import lab1.datalayer.FieldChecker;
import lab1.datalayer.InMemoryDataLayer;
import lab1.models.User;
import org.springframework.stereotype.Component;

import java.util.ArrayList;

@Component
public class UserService {

    private final DataLayer dataLayer;
    public UserService(){
        dataLayer = new InMemoryDataLayer();
    }

    public ArrayList<User> GetUsers(){
       return dataLayer.GetUsers();
    }

    public User GetUserById(long Id) throws IllegalArgumentException{
        User res = dataLayer.GetUser(Id);
        return res;
    }

    public boolean CreateUser(User user){
        if (FieldChecker.CheckUserFields(user)){
            dataLayer.CreateUser(user);
            return user.getId() != -1;
        }else
            return false;
    }

    public User DeleteUser(long Id){
        return dataLayer.DeleteUser(Id);
    }

    public User UpdateUser(User user){
        if (FieldChecker.CheckUserFields(user))
            return dataLayer.UpdateUser(user);
        else
            return null;
    }

    public DataLayer getDataLayer(){
        return dataLayer;
    }

}
